src/utils/metadata.ts

```diff
@@ -128,6 +130,73 @@ export function transformPublishFormToMetadata(
   return metadata
 }
 
+async function isDockerHubImageValid(
+  image: string,
+  tag: string
+): Promise<boolean> {
+  try {
+    const response = await axios.get(
+      `https://hub.docker.com/v2/repositories/${image}/tags/${tag}`
+    )
+    if (!response || response.status !== 200 || !response.data) {
+      toast.error(
+        'Could not fetch docker hub image info. Please check image name and tag and try again'
+      )
+      return false
+    }
+
+    return true
+  } catch (error) {
+    Logger.error(error.message)
+    toast.error(
+      'Could not fetch docker hub image info. Please check image name and tag and try again'
+    )
+    return false
+  }
+}
+
+async function is3rdPartyImageValid(imageURL: string): Promise<boolean> {
+  try {
+    const response = await axios.options(imageURL)
+    if (!response || response.status !== 200) {
+      toast.error(
+        'Could not fetch docker image info. Please check URL and try again'
+      )
+      return false
+    }
+    return true
+  } catch (error) {
+    Logger.error(error.message)
+    toast.error(
+      'Could not fetch docker image info. Please check URL and try again'
+    )
+    return false
+  }
+}
+
+function isUrl(image: string): boolean {
```

Comment:
we have a package [`is-url-superb`](https://github.com/sindresorhus/is-url-superb) in our dependencies, so we might be able to re-use that instead of maintaining our own patterns here